service/src/test/org/apache/hive/service/server/TestHS2HttpServerLDAP.java

```diff
@@ -0,0 +1,197 @@
+    } finally {
+      if (httpclient != null) {
+        httpclient.close();
+      }
+    }
+  }
+
+  @Test
+  public void testValidCredentialsWithRequestParameters() throws Exception {
+    CloseableHttpClient httpclient = null;
+    try {
+      CookieStore httpCookieStore = new BasicCookieStore();
+      HttpClientBuilder builder = HttpClientBuilder.create().setDefaultCookieStore(httpCookieStore);
+      httpclient = builder.build();
+
+      httpclient = HttpClients.createDefault();
+      HttpParams params = new BasicHttpParams();
+      params.setParameter(HttpAuthService.USERNAME_REQUEST_PARAM_NAME, VALID_USER);
+      params.setParameter(HttpAuthService.PASSWORD_REQUEST_PARAM_NAME, VALID_PASS);
+
+      HttpPost httpPost = new HttpPost("http://" + HOST + ":" + webUIPort + "/login");
+      httpPost.setParams(params);
+      httpclient.execute(httpPost);
+
+      Assert.assertFalse(isAuthorized(httpCookieStore.getCookies()));
+    } finally {
+      if (httpclient != null) {
+        httpclient.close();
+      }
+    }
+  }
+
+  @Test
+  public void testInvalidCredentialsWithRequestParameters() throws Exception {
+    CloseableHttpClient httpclient = null;
+    try {
+      CookieStore httpCookieStore = new BasicCookieStore();
+      HttpClientBuilder builder = HttpClientBuilder.create().setDefaultCookieStore(httpCookieStore);
+      httpclient = builder.build();
+
+      httpclient = HttpClients.createDefault();
+      HttpParams params = new BasicHttpParams();
+      params.setParameter(HttpAuthService.USERNAME_REQUEST_PARAM_NAME, INVALID_USER);
+      params.setParameter(HttpAuthService.PASSWORD_REQUEST_PARAM_NAME, INVALID_PASS);
+
+      HttpPost httpPost = new HttpPost("http://" + HOST + ":" + webUIPort + "/login");
+      httpPost.setParams(params);
+      httpclient.execute(httpPost);
+
+      Assert.assertFalse(isAuthorized(httpCookieStore.getCookies()));
+    } finally {
+      if (httpclient != null) {
+        httpclient.close();
+      }
+    }
+  }
+
+  public static boolean isAuthorized(List<Cookie> cookies) {
+    Optional<Cookie> cookie = cookies.stream()
+        .filter(c -> c.getName().equals(LdapAuthService.HIVE_SERVER2_WEBUI_AUTH_COOKIE_NAME))
+        .findAny();
+    
+    if (!cookie.isPresent()) {
+      return false;
+    }
+
+    String signedCookieValue = cookie.get().getValue();
+    String cookieValue = hiveServer2.getLdapAuthService().verifyAndExtract(signedCookieValue);
+    String userName = HttpAuthUtils.getUserNameFromCookieToken(cookieValue);
+
+    return userName.equals(VALID_USER);
+  }
+  
+  @AfterClass public static void afterTests() {
+    hiveServer2.stop();
+  }
+  
+  public static class DummyLdapAuthenticationProviderImpl implements PasswdAuthenticationProvider {
+    @Override
+    public void Authenticate(String user, String password) throws AuthenticationException {
+      if (!user.equals(VALID_USER) && password.equals(VALID_PASS))
```

Comment:
is this condition intentional? I think it wants to be:
```
      if (!(user.equals(VALID_USER) && password.equals(VALID_PASS)))
```

Reply:
Done

Reply:
thanks